url/Slug.ts

```diff
@@ -1,27 +1,16 @@
+export type SlugCharacterReplacementMap = Array<[RegExp|string, string]>;
 
-
-export default class Slug
+export class Slug
 {
+    private readonly transforms : SlugCharacterReplacementMap;
     /**
-     * Constructs a new instance
-     *
-     * @param {Array.<Array.<*>>} transforms
-     * @param {?RegExp} sanitize
+     * The sanitizing regexp, that removes all unwanted characters
      */
-    constructor (transforms = [], sanitize = null)
+    private readonly sanitize : RegExp;
+
+    constructor (transforms : SlugCharacterReplacementMap = [], sanitize : null|RegExp= null)
     {
-        /**
-         * @private
-         * @type {Array.<Array.<RegExp, string>>}
-         */
         this.transforms = [];
-
-        /**
-         * The sanitizing regexp, that removes all unwanted characters
-         *
-         * @private
-         * @type {RegExp}
-         */
         this.sanitize = null !== sanitize
```

Comment:
`private`?

Reply:
```suggestion
        private this.sanitize = null !== sanitize
```

Reply:
At the declaration level, not at the assignment :)